core/src/main/java/org/infinispan/configuration/cache/IndexingConfigurationBuilder.java

```diff
@@ -337,11 +349,18 @@ public IndexingConfiguration create() {
    @Override
    public IndexingConfigurationBuilder read(IndexingConfiguration template) {
       attributes.read(template.attributes());
+      Index index = attributes.attribute(INDEX).get();
```

Comment:
This addition ensures inheritance works properly when inheriting from an old base config that uses ```index`` or ```auto-config``` attributes rather than ```enabled```.  Do you think it's worth adding this explanation in code @gustavonalle ?